diff --git a/smtp.mail.inc b/smtp.mail.inc
index b064330..6c5f954 100644
--- a/smtp.mail.inc
+++ b/smtp.mail.inc
@@ -779,7 +779,7 @@ class SmtpMailSystem implements MailSystemInterface {
     //  some name <address@example.com>
     //  "another name" <address@example.com>
     //  <address@example.com>
-    if (preg_match('/^"?([^"\t\n]*)"?\s*<([^>\t\n]*)>$/', $input, $matches)) {
+    if (preg_match('/^"?([^"]*)"?\s*<(.*)>$/', $input, $matches)) {
       $components['name'] = trim($matches[1]);
       $components['email'] = trim($matches[2]);
     }
